apps/ybribe/utils/actions.ts

```diff
@@ -0,0 +1,161 @@
+		});
+		const {hash} = await writeContract(config.request);
+		const receipt = await waitForTransaction({chainId: wagmiProvider.chainId, hash});
+		if (receipt.status === 'success') {
+			props.statusHandler?.({...defaultTxStatus, success: true});
+		} else if (receipt.status === 'reverted') {
+			props.statusHandler?.({...defaultTxStatus, error: true});
+		}
+		return ({isSuccessful: receipt.status === 'success', receipt});
+	} catch (error) {
+		console.error(error);
+		const errorAsBaseError = error as BaseError;
+		captureException(errorAsBaseError);
+		props.statusHandler?.({...defaultTxStatus, error: true});
+		return ({isSuccessful: false, error: errorAsBaseError || ''});
+	} finally {
+		setTimeout((): void => {
+			props.statusHandler?.({...defaultTxStatus});
+		}, 3000);
+	}
+
+}
+
+export async function claimRewardV3(props: TClaimReward): Promise<TTxResponse> {
+	assertAddress(CURVE_BRIBE_V3_ADDRESS);
+	assertAddress(props.contractAddress);
+	assertAddress(props.gaugeAddress);
+	assertAddress(props.tokenAddress);
+
+	props.statusHandler?.({...defaultTxStatus, pending: true});
+	const wagmiProvider = await toWagmiProvider(props.connector);
+
+	try {
+		const config = await prepareWriteContract({
+			...wagmiProvider,
+			address: toWagmiAddress(CURVE_BRIBE_V3_ADDRESS),
+			abi: CURVE_BRIBE_V3_ABI,
+			functionName: 'claim_reward_for',
+			args: [wagmiProvider.address, props.gaugeAddress, props.tokenAddress]
+		});
+		const {hash} = await writeContract(config.request);
+		const receipt = await waitForTransaction({chainId: wagmiProvider.chainId, hash});
+		if (receipt.status === 'success') {
+			props.statusHandler?.({...defaultTxStatus, success: true});
+		} else if (receipt.status === 'reverted') {
+			props.statusHandler?.({...defaultTxStatus, error: true});
+		}
+		return ({isSuccessful: receipt.status === 'success', receipt});
+	} catch (error) {
+		console.error(error);
+		const errorAsBaseError = error as BaseError;
+		captureException(errorAsBaseError);
+		props.statusHandler?.({...defaultTxStatus, error: true});
+		return ({isSuccessful: false, error: errorAsBaseError || ''});
+	} finally {
+		setTimeout((): void => {
+			props.statusHandler?.({...defaultTxStatus});
+		}, 3000);
+	}
+
+}
+
+
+/* 🔵 - Yearn Finance **********************************************************
+** claimReward is a _WRITE_ function that claims the rewards from the yBribe
+** contract.
+** The correct function for V2 or V3 should be used.
+**
+** @app - yBribe
+** @param gaugeAddress - The address of the gauge to claim rewards from.
+** @param tokenAddress - The address of the token to claim rewards from.
+******************************************************************************/
+type TAddReward = TWriteTransaction & {
+	gaugeAddress: TAddressWagmi;
+	tokenAddress: TAddressWagmi;
+	amount: bigint;
+};
+export async function addReward(props: TAddReward): Promise<TTxResponse> {
+	assertAddress(CURVE_BRIBE_V3_ADDRESS);
```

Comment:
here

I see a lot of duplication, was it here that the idea of having this refactored later on?